lib/spack/spack/paths.py

```diff
@@ -50,7 +51,13 @@
 mock_packages_path = os.path.join(repos_path, "builtin.mock")
 
 #: User configuration location
-user_config_path = os.path.expanduser('~/.spack')
+user_config_path = re.sub(r'\$(?:spack\b|\{spack\})',
+                          spack_root,
+                          os.path.expanduser(
+                              os.environ.get(
+                                  'SPACK_USER_CONFIG',
+                                  '~/.spack')))
```

Comment:
Isn't it better to make an empty value default to `~/.spack` too?

`os.environ.get('SPACK_USER_CONFIG') or '~/.spack'`.

Also I would prefer `SPACK_USER_CONFIG_PATH`. It's super long, but it can only mean one thing.

Reply:
I prefer `SPACK_USER_CONFIG_PATH` as well.

As to making  an empty value default to `~/.spack` - I suppose if the user took the effort to `export SPACK_USER_CONFIG_PATH=` they did so for a reason?

Reply:
To some people that means the variable was "unset". I'm not very much attached to that opinion though